test/models/spot_test.rb

```diff
@@ -0,0 +1,340 @@
+require 'test_helper'
+
+# rubocop:disable Metrics/ClassLength
+class SpotTest < ActiveSupport::TestCase
+  test 'spot must have row' do
+    spot = build(:spot, row: nil)
+    refute spot.valid?
+  end
+
+  test 'spot must have file' do
+    spot = build(:spot, file: nil)
+    refute spot.valid?
+  end
+
+  test 'a and x rows should allow efer, solo, first trumpet parts' do
```

Comment:
this test can be combined with `test 'a and x rows shouldn't allow non efer, solo, first parts`. That would cut this diff down by 1/3. See `user_test.rb`